### PR TITLE
Added --serializer option to cli, finishing off PR #1381 and fix #1357

### LINKED ISSUE #1357: Job picked from queue with default serializer while worker has json serializer
I configured the worker and queue with `serializer=json` but when the job is picked from the queue it tries to de-serialize with pickle and I get pickle errors.

I added some logging: At startup you can see the json serializer, but when the job is processed, the queue and job have a default serializer.

```
Starting Task Worker (pid: 13882)
-- Queue serializer <module 'json' from '/usr/lib/python3.8/json/__init__.py'>
-- Worker serializer <module 'json' from '/usr/lib/python3.8/json/__init__.py'>
16:04:47 Registering birth of worker python-consumer
16:04:47 Worker rq:worker:python-consumer: started, version 1.5.2
16:04:47 *** Listening on queue...
16:04:47 Sent heartbeat to prevent worker timeout. Next one should arrive within 480 seconds.
16:04:47 *** Listening on queue...
16:04:47 Sent heartbeat to prevent worker timeout. Next one should arrive within 480 seconds.
-- Queue serializer <class 'rq.serializers.DefaultSerializer'>
-- Job serializer <class 'rq.serializers.DefaultSerializer'>
16:04:56 queue: Export (52bb0e76-4387-4e46-8fab-f462ce54914c)
16:04:56 Sent heartbeat to prevent worker timeout. Next one should arrive within 480 seconds.
16:04:56 Sent heartbeat to prevent worker timeout. Next one should arrive within 90 seconds.
-- Job serializer <class 'rq.serializers.DefaultSerializer'>
16:04:56 Handling failed execution of job 52bb0e76-4387-4e46-8fab-f462ce54914c
-- Job serializer <class 'rq.serializers.DefaultSerializer'>
16:04:56 Sent heartbeat to prevent worker timeout. Next one should arrive within 480 seconds.
16:04:56 Sent heartbeat to prevent worker timeout. Next one should arrive within 480 seconds.
16:04:56 *** Listening on queue...
16:04:56 Sent heartbeat to prevent worker timeout. Next one should arrive within 480 seconds. 
```

I could fix it by passing the `serializer=self.serializer` form the worker to the queue and job via the `dequeue_any` class method in `dequeue_job_and-maintain_ttl`.

Is this a bug with the json serializer, or was my setup wrong?

### PR DESCRIPTION
**Pull request by f0cker**

### CODE REVIEW

File: rq/cli/cli.py [hunk 1]
```diff
@@ -62,7 +62,10 @@
     click.option('--path', '-P',
                  default='.',
                  help='Specify the import path.',
-                 multiple=True)
+                 multiple=True),
+    click.option('--serializer', '-S',
+                 default=None,
+                 help='Serializer to use, JSON or Pickle')
```
Comment (selwin) [inline rq/cli/cli.py:68]: Can we clarify this a bit to `Path to serializer, defaults to rq.serializers.DefaultSerializer`?

**Comment by selwin** [comment]
Do you mind documenting this in https://github.com/rq/rq/blob/master/docs/docs/workers.md ?

**Comment by f0cker** [comment]
Done, the default setting and documentation. Thanks.

File: rq/cli/cli.py (hunk 1 above)
Comment (selwin) [reply rq/cli/cli.py:70]: Mind addressing this comment?

File: docs/docs/workers.md [hunk 2]
```diff
@@ -69,6 +69,7 @@ In addition to `--burst`, `rq worker` also accepts these arguments:
 * `--date-format`: Datetime format for the worker logs, defaults to `'%H:%M:%S'`
 * `--disable-job-desc-logging`: Turn off job description logging.
 * `--max-jobs`: Maximum number of jobs to execute.
+* `--serializer`: Serialization to use, i.e. Pickle, JSON or other custom serializer
```
Comment (selwin) [inline docs/docs/workers.md:72]: Can we clarify that this is a path to serializer object? Something like `Path to serializer (e.g  "rq.serializers.DefaultSerializer" or "rq.serializers.JSONSerializer"`

Comment (f0cker) [reply docs/docs/workers.md]: Done.

File: rq/cli/cli.py (hunk 1 above)
Comment (selwin) [reply rq/cli/cli.py:70]: This help message should also be changed @f0cker . Apart from this, I think this PR is ready to merge :)

Comment (f0cker) [reply rq/cli/cli.py]: Done. Let me know if you need anything else.

**Comment by selwin** [comment]
Thanks!

**Pull request merged**